### PR TITLE
WIP python-sgd-regression

### PR DESCRIPTION
**Pull request by Marigold**
First attempt at distributed python-sgd-regression.

See old comments at https://github.com/HBPMedical/algorithm-repository/pull/4

### CODE REVIEW

**Comment by ludovicc** [comment]
I have added a couple of commits into Marigold-python-sgd-regression, trying to execute the tests

**Comment by Marigold** [comment]
@ludovicc `captain test` is not passing yet, I have to change sample data a bit to include mean & std of variables and also split the `main` method to `compute intermediate` and `compute final` (because of removing that INSERT ON UPDATE from `io_helper`). Will try to get it done asap

**Comment by Marigold** [comment]
@ludovicc integration test is now passing and I split it into two modes - one for `partial_fit` and second for generating PFA code. The orchestration would have to be handled by woken

File: python-sgd-regression/sgd_regression.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        elif model_type == 'naive_bayes':
+            estimator = MixedNB(**model_parameters)
+        else:
+            raise ValueError('Unknown model type {} for classification'.format(model_type))
+
+    return estimator
+
+
+def get_Xy(dep_var, indep_vars):
+    """Create dataframe from input data.
```
[the remark this answers is not in the document]
Comment (Marigold) [reply python-sgd-regression/sgd_regression.py]: That's exactly what I was planning to do, it's just that `get_Xy` each ML method looks always a bit different, so I was hoping to unify it into single function as part of some refactoring.

IMHO `fetch_data` could also be improved to fit out latest requirements (labels, etc.). I think it shouldn't modify original metadata that much and just return everything (labels, descriptions, etc.).

File: python-sgd-regression/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,34 @@
+FROM hbpmip/python-mip:0.2.4
+
+MAINTAINER mirco.nasuti@chuv.ch
+
+ENV DOCKER_IMAGE=hbpmip/python-sgd-regression:0.0.7 \
+    FUNCTION=python-sgd-regression
+
+# TODO: git is required to install `git+https://github.com/jsonpickle/jsonpickle.git` for scikit-learn model
+# serialization - we can remove it once jsonpickle==0.9.6 is available on PyPi
```
[the remark this answers is not in the document]
Comment (Marigold) [reply python-sgd-regression/Dockerfile]: Fix coming in next PR

**Pull request merged**